src/calliope/preprocess/data_sources.py

```diff
@@ -245,7 +245,8 @@ def _read_csv(self) -> pd.DataFrame:
             pd.DataFrame: Loaded data without any processing.
         """
         filename = self.input["source"]
-        csv_reader_kwargs = {"header": self.columns, "index_col": self.index}
+        header = [0] if self.columns is None else self.columns
```

Comment:
I am a bit confused by your comment on this line (in your  Would this generate a weird error for users in certain situations, or avoid it?

Since Calliope making an assumption here, maybe just throw a warning?

Reply:
Honestly, I think the most robust approach is to just ask users to fully fill it by default, and to ensure it has no numbers in it on our end if we set it to 0. It reduces the number of test cases on our end, leads to better documentation, and may avoid headaches in the future. 

Unless I am missing something?

Reply:
Sadly, there are legitimate reasons to have your dimensions or their values be numeric (e.g. years of data), so we can't set that strict rule.

Reply:
> I am a bit confused by your comment on this line (in your Would this generate a weird error for users in certain situations, or avoid it?

The issue with pandas Series/DataFrames and their CSV representation is that column names and MultiIndex index levels are stored on the same CSV row. It gets even weirder with MultiIndex columns. If you don't define any dimension linked to your non-index data, you still have a header row in the CSV in which your `row` dimensions are stored (MultiIndex levels). Since we're forcing that a user explicitly defines in their CSV header any dimensions that are in `rows`/`columns` in their data source definition, we always take it that there is at least one header row. If there are multiple `columns` dimensions then those will be inferred from the list the user provides.

Reply:
Go it. This one is hard for sure. I am satisfied with the improved error handling.
We can revisit this if it's causing too many issues, although I am not sure how to solve it.